akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/GuardianStartupBehavior.scala

```diff
@@ -33,17 +32,19 @@ private[akka] final class GuardianStartupBehavior[T](val guardianBehavior: Behav
 
   import GuardianStartupBehavior.Start
 
-  private val stash = StashBuffer[Any](1000)
+  private var tempStash: List[Any] = Nil
 
   override def onMessage(msg: Any): Behavior[Any] =
     msg match {
       case Start =>
         // ctx is not available initially so we cannot use it until here
-        Behaviors.setup(ctx =>
-          stash
-            .unstashAll(ctx, Behaviors.intercept(() => new GuardianStopInterceptor)(guardianBehavior.unsafeCast[Any])))
+        Behaviors.withStash[Any](1000, stash => {
+          tempStash.reverse.foreach(stash.stash)
+          tempStash = null
+          stash.unstashAll(Behaviors.intercept(() => new GuardianStopInterceptor)(guardianBehavior.unsafeCast[Any]))
+        })
       case other =>
-        stash.stash(other)
+        tempStash = other :: tempStash
```

Comment:
Not likely a problem, but size check here?

Reply:
I'll add it in and throw an exception so it is obvious what is wrong rather than an OOM :)